internal/getmodules/installer.go

```diff
@@ -65,7 +70,15 @@ func NewPackageFetcher(env PackageFetcherEnvironment) *PackageFetcher {
 // caller must resolve that itself, possibly with the help of the
 // getmodules.SplitPackageSubdir and getmodules.ExpandSubdirGlobs functions.
 func (f *PackageFetcher) FetchPackage(ctx context.Context, instDir string, packageAddr string) error {
-	return f.getter.getWithGoGetter(ctx, instDir, packageAddr)
+	ctx, span := tracing.Tracer().Start(ctx, "Fetch Package",
+		trace.WithAttributes(semconv.URLFull(packageAddr)),
+	)
+	defer span.End()
+	err := f.getter.getWithGoGetter(ctx, instDir, packageAddr)
+	if err != nil {
+		span.RecordError(err)
+	}
+	return nil
```

Comment:
Was this on purpose? Or just a slip up?
```suggestion
	return err
```

Reply:
Infact, What I should've done here is return err inside the `!= nil` check. good catch!

Reply:
Fixed in https://github.com/opentofu/opentofu/pull/2711/commits/3be26199485c5b72d1b0b9c0144282df57e01de8